storage/src/main/java/org/apache/uniffle/storage/handler/impl/LocalFileServerReadHandler.java

```diff
@@ -80,7 +81,7 @@ private void prepareFilePath(
     File baseFolder = new File(fullShufflePath);
     if (!baseFolder.exists()) {
       // the partition doesn't exist in this base folder, skip
-      throw new RuntimeException("Can't find folder " + fullShufflePath);
+      throw new FileNotFoundException("Can't find folder " + fullShufflePath);
```

Comment:
Change exception to `FileNotFoundException` ，this should not be thrown to client. 

For example:
When using the MEMORY_LOCALFILE_HDFS type, sometime, the localfile storage may be empty due to the big event flushed to HDFS directly. So when I want to fast fail in `ComposedClientReadHandler`, this problem should be involved. So this exception should be handled in server's grpc api and return the empty index result.